Signup: Use --color-primary for the background in signup

shaunandrews opened the pull request:
This brings us closer to our prototype.

![signup new colors](https://user-images.githubusercontent.com/191598/50861248-13b91780-1366-11e9-836c-86e8afb3c842.gif)

fditrapani commented:
This is looking really nice! Some minor things I spotted:

- The tabs on the plans page look broken. 
- The "Awesome! Give us one minute and we’ll move right along." copy doesn't have enough contrast against the background (it's black copy on dark blue).
- I see a weird ghosting banding pattern but it doesn't appear when I take screenshots. Does anyone else see it?

> Just something I'm personally curious about - how do we keep CSS variables working in older browsers?

Good question @adamziel. Not that this should be a blocker for this PR but why we would use CSS variables when we could uses SASS variables and not worry about support?

flootr commented:
> Just something I'm personally curious about - how do we keep CSS variables working in older browsers?

We use a PostCSS plugin (`postcss-custom-properties`) which takes care of adding the necessary fallbacks.

> [...] why we would use CSS variables when we could uses SASS variables and not worry about support?

At least my main concern is consistency with how we use colors in Calypso in the future. We'll be shortly publishing a p2 about that exact topic 🙂  /cc @drw158

fditrapani commented:
Thanks for the extra context @flootr. Sounds good 👍

sixhours commented:
The Domains step needs a few tweaks (in case this color change launches before we remove the Domains step; if not, please disregard):

Some links are hidden until you hover over them, and then the contrast is still poor:
<img width="289" alt="screen shot 2019-01-09 at 1 52 06 pm" src="https://user-images.githubusercontent.com/2124984/50921846-fabc6f00-1416-11e9-9147-45decc1e72d9.png">

Similar contrast issues at the bottom of the Transfer/Map domains screen:
<img width="732" alt="screen shot 2019-01-09 at 1 52 30 pm" src="https://user-images.githubusercontent.com/2124984/50921931-2b9ca400-1417-11e9-81df-728067cb49ab.png">

Funky gradients around this graphic:
<img width="621" alt="screen shot 2019-01-09 at 1 50 32 pm" src="https://user-images.githubusercontent.com/2124984/50921854-ff812300-1416-11e9-8363-29fde9645c42.png">

Didn't we fix this nonsense?!? I feel like we did, but I can open a separate PR:
<img width="1280" alt="screen shot 2019-01-09 at 1 53 19 pm" src="https://user-images.githubusercontent.com/2124984/50921868-05770400-1417-11e9-9dc7-48ec06c6ce63.png">

shaunandrews commented:
All issues above should be fixed.

fditrapani commented:
This is looking really good @shaunandrews 👊

fditrapani commented:
Oh wait.. I just noticed some text needs updating on `/start/import`

![image](https://user-images.githubusercontent.com/6981253/51346014-87e07300-1a6a-11e9-86fb-706b1614df87.png)

shaunandrews commented:
Nice catch, @fditrapani — I pushed a few updates. Check again, please

<img width="1183" alt="image" src="https://user-images.githubusercontent.com/191598/51348313-66828580-1a70-11e9-8d4a-82c50a5c7096.png">

fditrapani commented:
Thanks @shaunandrews. This is looking really good! I visited all the flows on [this funnel](https://mc.a8c.com/tracks/funnel/view/?15733/2018-12-15/to/2019-01-13&selected-segmentation%5B0%5D%5Btype%5D=custom_prop&selected-segmentation%5B0%5D%5Bstep%5D=0&selected-segmentation%5B0%5D%5Bprop%5D=flow&) and only found a couple things. Some unrelated cause I was just in there but here you go anyways :)

**If you encounter an error, this is what you see:**
<img width="1625" alt="screen shot 2019-01-17 at 8 14 35 pm" src="https://user-images.githubusercontent.com/6981253/51359446-cd666580-1a95-11e9-9a28-0396ee2dbe3e.png">

I got here by visiting `/start/site-selected`

**I'm not a fan of the grey on blue when the screen initially loads:**

<img width="1625" alt="screen shot 2019-01-17 at 8 01 53 pm" src="https://user-images.githubusercontent.com/6981253/51359592-77de8880-1a96-11e9-8c12-aeb9e217a816.png">

I think I'd like it better if it were dark blue like on our home page. 

**Not related to colours, The logo still looks super huge to me here:**
<img width="1625" alt="screen shot 2019-01-17 at 8 05 15 pm" src="https://user-images.githubusercontent.com/6981253/51359628-993f7480-1a96-11e9-9b79-273d514e3e02.png">

**Also not related to this PR, I [found a flow](https://wordpress.com/start/subdomain) that looks really broken:**
<img width="1625" alt="screen shot 2019-01-17 at 8 26 28 pm" src="https://user-images.githubusercontent.com/6981253/51359533-3057fc80-1a96-11e9-96b7-345179948165.png">

The Pressable part at the top should probably go. I got here by visiting `/start/subdomain`

michaeldcain commented:
> Also not related to this PR, I found a flow that looks really broken:

Let's just remove that in a separate PR - no need to fix the styles.

shaunandrews commented:
> Let's just remove that in a separate PR - no need to fix the styles.

Sounds good. I haven't touched that flow.

> I'm not a fan of the grey on blue when the screen initially loads

Fixed.

<img width="260" alt="screen shot 2019-01-18 at 10 57 00 am" src="https://user-images.githubusercontent.com/191598/51397788-f6c3d780-1b0f-11e9-9399-28060edabc7c.png">

> If you encounter an error

<img width="825" alt="screen shot 2019-01-18 at 10 56 53 am" src="https://user-images.githubusercontent.com/191598/51397817-05aa8a00-1b10-11e9-9431-f6a10d87f3ae.png">

Fixed.

fditrapani commented:
Looking good @shaunandrews. Thanks for the updates.

scruffian commented:
This breaks the Woocommerce WPCC connect flow:
<img width="1027" alt="screenshot 2019-01-18 at 16 54 02" src="https://user-images.githubusercontent.com/275961/51401040-c1d07880-1b41-11e9-97f2-b5bfc549bb15.png">

To reproduce this:
- Go to https://woocommerce.com/
- Click "Log in with WordPress.com"
- Click on "Create an account"
- Update wordpress.com in the URL to calypso.localhost:300

mattwiebe commented:
Found some ugly in the site style button row atop http://calypso.localhost:3000/start/onboarding-dev/site-style-with-preview

<img width="982" alt="image" src="https://user-images.githubusercontent.com/195089/51402412-0d1f6280-1b13-11e9-8aac-0463c57f57be.png">

shaunandrews commented:
> Found some ugly in the site style button row

Confirmed. This space intentionally left ugly. For now at least. Its only displayed in `onboarding-dev` — once we have real style variations to display (more than just font changes) we'll give this screen the real love it deserves.

mattwiebe commented:
Just as an FYI you can find enabled flows in `signup/config/flows-pure.js` - I went through there and loaded flows with steps that haven't been covered yet. (We should really make a way to comb through here and find a definitive list programmatically.)

http://calypso.localhost:3000/start/pressable-nux has horrid contrast

<img width="676" alt="image" src="https://user-images.githubusercontent.com/195089/51403295-a8b1d280-1b15-11e9-9b83-296c9307c54c.png">

http://calypso.localhost:3000/start/rewind-switch/rewind-migrate - text at the bottom, top should be similar to the above

<img width="760" alt="image" src="https://user-images.githubusercontent.com/195089/51403431-0e9e5a00-1b16-11e9-8421-1316bb31f14c.png">

http://calypso.localhost:3000/start/rewind-switch/rewind-migrate/rewind-setup - similar again to above

<img width="760" alt="image" src="https://user-images.githubusercontent.com/195089/51403495-3db4cb80-1b16-11e9-9ea8-bb105e027466.png">

http://calypso.localhost:3000/start/reader woof

<img width="1093" alt="image" src="https://user-images.githubusercontent.com/195089/51403778-190d2380-1b17-11e9-94f4-1b226a2b7bc9.png">

shaunandrews commented:
Found another one:

<img width="1395" alt="screen shot 2019-01-18 at 2 33 14 pm" src="https://user-images.githubusercontent.com/191598/51408862-0b16cd00-1b2e-11e9-971e-9a1fa5d91717.png">

`/start/crowdsignal/`

shaunandrews commented:
I pushed some updates to fix the Crowdsignal and Woocommerce oauth flows.

shaunandrews commented:
Pushed an update to fix the `/start/reader` flow:

<img width="1031" alt="image" src="https://user-images.githubusercontent.com/191598/51412761-b3329300-1b3a-11e9-9af3-1934e3deca84.png">

shaunandrews commented:
Pushed another update to fix all the screens @mattwiebe pointed out above.

southp commented on client/components/domains/example-domain-browser/index.jsx:
```diff
@@ -11,20 +11,21 @@ export default function ExampleDomainBrowser( { className } ) {
-						d="M150.336 23.08c1.252 0 2.087.92 2.117 2.32h-4.35c.094-1.4.973-2.32 2.233-2.32zm2.08 4.877c-.33.695-1.018 1.07-2.036 1.07-1.34 0-2.212-.99-2.278-2.55v-.06h5.676v-.482c0-2.454-1.296-3.97-3.427-3.97-2.167 0-3.56 1.612-3.56 4.095 0 2.497 1.37 4.08 3.56 4.08 1.73 0 2.945-.828 3.326-2.183h-1.26zm6.138-.938L160.42 30h1.496l-2.718-4 2.68-3.896h-1.427l-1.852 2.938h-.118l-1.875-2.938h-1.5l2.73 3.9 ...
-						fill="#3D596D"
-					/>
+				<g fill="none" fill-rule="evenodd">
```
It's weird, but `react` requires this to be `fillRule` or it generates warning:
![image](https://user-images.githubusercontent.com/1842898/51456327-34cb3080-1d88-11e9-8507-177562943260.png)

shaunandrews replied:
Oh, `react`...

southp commented on client/components/domains/example-domain-browser/index.jsx:
```diff
@@ -11,20 +11,21 @@ export default function ExampleDomainBrowser( { className } ) {
+				<g fill="none" fill-rule="evenodd">
+					<path fill="#D8D8D8" d="M10 0h285v50H0V10C0 4.477 4.477 0 10 0z" />
+					<path fill="#FFF" d="M0 50h295v50H0zM94 9h201v30H94a4 4 0 0 1-4-4V13a4 4 0 0 1 4-4z" />
+					<text fill="#D8D8D8" font-family="SFUIText-Regular, SF UI Text" font-size="14">
```
Same thing here. `react` wants `fontFamily` and `fontSize` instead.

southp commented on client/components/domains/example-domain-browser/index.jsx:
```diff
@@ -11,20 +11,21 @@ export default function ExampleDomainBrowser( { className } ) {
+							https://
+						</tspan>
+					</text>
+					<text fill="#424242" font-family="SFUIText-Regular, SF UI Text" font-size="14">
```
See above comment.

shaunandrews commented:
> First, clicking "signup" from devdocs/start results in missing background color:

I'm not sure what's causing thing — if you visit `/start/developer` directly all is fine. If you click the buttons from `/devdocs/start` it doesn't update the body classes correctly. This is worth looking into, but I don't think its a blocker for this PR.

I'm working on updates the Akismet WPCC flow. Thanks for that catch.

michaeldcain commented:
Found the following when choosing to map a domain I already own from the `/domains/` step:

![screen shot 2019-01-21 at 3 33 19 pm](https://user-images.githubusercontent.com/942359/51499093-e8d3b800-1d96-11e9-8f5f-99be15763898.png)

shaunandrews commented:
Text below the `'use-your-domain` screen should be fixed now.

<img width="521" alt="image" src="https://user-images.githubusercontent.com/191598/51499373-06555180-1d98-11e9-8207-291c5e6f7e9c.png">
